### PR TITLE
Connector Builder: assert catalog exists on `test_read` requests

### PR DESCRIPTION
**Pull request by clnoll**
Adds a check that the catalog exists for `test_read` requests.

### CODE REVIEW

File: airbyte-cdk/python/connector_builder/main.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ def handle_connector_builder_request(
     if command == "resolve_manifest":
         return resolve_manifest(source)
     elif command == "test_read":
+        assert catalog is not None, "`test_read` requires a valid `ConfiguredAirbyteCatalog`, got None."
```
Comment (maxi297) [inline airbyte-cdk/python/connector_builder/main.py:55]: This would be part of the API between the Connector Builder and the CDK. Is this what we expect?

Comment (clnoll) [reply airbyte-cdk/python/connector_builder/main.py:55]: Yes, it ultimately gets wrapped in an AirbyteTraceMessage, which is handled by the connector builder server.

**Review by maxi297 — APPROVED** [review]
LGTM!

**Pull request merged**